File: EngagementHub/TOS.EngagementHub.Models/Filters/EmployeeFilter.cs
using TOS.Common.DataModel;

namespace TOS.EngagementHub.Models.Filters
{
    public class EmployeeFilter : QueryInput
    {
        public EmployeeFilter(string nameContains = "", int offset = -1, int limit = -1, string sortBy = null, SortDirection sortDirection = SortDirection.Asc)
            : base(offset, limit, sortBy, sortDirection)
        {
            NameContains = nameContains;
        }

        public string NameContains { get; set; }
    }
}

File: NiceReads/Backend/Server/tos-server/src/application/models/filters/EmployeeFilter.ts
import { QueryInput } from '../../../common/datamodel/QueryInput';

export class EmployeeFilter extends QueryInput {
    constructor(
        public readonly nameContains: string = '',
        offset: number = -1,
        limit: number = -1,
        sortBy: string | undefined = undefined,
        sortDirection: SortDirection = SortDirection.ASC
    ) {
        super(offset, limit, sortBy, sortDirection);
    }
}